source/common/uma_pools/usm_allocator_config.hpp

```diff
@@ -0,0 +1,34 @@
+//===--- usm_allocator_config.hpp -configuration for USM memory allocator---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef USM_ALLOCATOR_CONFIG
+#define USM_ALLOCATOR_CONFIG
+
+#include "usm_allocator.hpp"
+
+#include <string>
+
+namespace usm_settings {
+
+enum MemType { Host, Device, Shared, SharedReadOnly, All };
+
+// Parses and stores configuration for all instances of USM allocator
+class USMAllocatorConfig {
+  public:
+    USMAllocatorConfig(const std::string &config = "", bool trace = true);
+
+    size_t EnableBuffers = 1;
+    USMAllocatorParameters Configs[MemType::All];
+
+    // String names of memory types for printing in limits traces.
+    static constexpr const char *MemTypeNames[MemType::All] = {
```

Comment:
does this need to be in the header? I think you can get the same information from `Configs[MemType::whatever].memoryTypeName`.

Reply:
Well, this is just mapping from enum to string used to initialize the `Configs[MemType::whatever].memoryTypeName` - I can put it into usm_allocator_config.cpp. I did not want to do too many changes in this PR but eventually, I'd like to get rid of the `enum MemType` at all and use some UR type.

Reply:
Removed.